### PR TITLE
Turretcide

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
Balance "Nerfed aboveground turret presence and ammo availability."

#### Purpose of change
Excessive ammunition availability from turrets has been a balance problem for a while now, especially since aboveground turrets are really easy to cheese.
Having these things deployed in large numbers was also a bad match for the thematics of the game. Even if someone decided it was a good idea, rolling them out in large numbers would be logistically prohibitive.
In terms of world consistency, having these turrets "in stasis" until the player approaches is problematic because the surface is otherwise very dynamic and it's common to encounter a situation where a turret *should* have it's ammo depleted or at least exhausted long before the player arrives.

#### Describe the solution
Eliminate all the military themed spawns of turrets aboveground.
The only remaining aboveground turret spawns are turrets scavenged by some NPCs, but those have an alternate fix which is nerfing their ammo count when spawning.

#### Describe alternatives you've considered
Nerfing all turret ammo spawn numbers has been pointed out, but doesn't handle the consistency goal.
If the turrets simply have very low ammo stores, and they're inconsistent with the IRL turrets they're based on.
If the turrets are spawned with "expended ammunition" around them, they run the risk of having no nearby corpses to reflect the targets that caused the ammo to be expended.
If people wanted to craft maps where the turrets had been overrun or partially overrun that would be fine.
I kind of want a system where the turrets can dynamically interact with zombie hordes and eventually get overrun, but that's pretty far off, we can just re-add them later.

#### Testing
Spawned some of the locations to glance them over.
Verified that isolated artisans turrets spawn with the ammunition override applying properly.

### CODE REVIEW

**Comment by anoobindisguise** [comment]
This is great, big steps towards fixing the very silly turret spawns in the world.
What do you want to do about the 4 turrets that spawn in the TCL lobby + the 6-odd more that spawn in other parts of the facility or the 6-7 that spawn in military bunkers?

**Comment by kevingranade** [comment]
Leave them for now, because they aren't as trivial to cheese with e.g. vehicle ramming, and they don't spawn as frequently.

File: data/json/mapgen/helipad.json [hunk 1]
```diff
@@ -117,7 +117,6 @@
       },
       "vendingmachines": { "1": { "item_group": "vending_drink", "lootable": true }, "2": { "item_group": "vending_food", "lootable": true } },
       "place_monsters": [ { "monster": "GROUP_MIL_STRONG", "x": 36, "y": 15 }, { "monster": "GROUP_MIL_WEAK", "x": 24, "y": 30, "density": 2 } ],
-      "monster": { "T": { "monster": "mon_turret_rifle" } },
```
Comment (Maleclypse) [inline data/json/mapgen/helipad.json:120]: Would it make sense to change these to spotlight turrets?

**Comment by Inglonias** [comment]
Does this also apply to riot control turrets?

**Comment by Wishbringer** [comment]
> Does this also apply to riot control turrets?

No, does not seem like it.

**Comment by SariusSkelrets** [comment]
Riot control turrets only shot beanbags and spawn with a few of them. They don't give hundreds of firearm ammo

File: data/json/mapgen/helipad.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
       "fill_ter": "t_floor",
       "rows": [
         "FFF.......pppppppp...........................FFF",
-        "FTFFWWWWWF!!!!!!!!FFFFFFFFFFFFFFFFFFFFFFFFFFFFTF",
+        "FFFFWWWWWF!!!!!!!!FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
```
Comment (kevingranade) [inline data/json/mapgen/helipad.json:11]:
```suggestion
        "FTFFWWWWWF!!!!!!!!FFFFFFFFFFFFFFFFFFFFFFFFFFFFTF",
```

File: data/json/mapgen/helipad.json [hunk 3]
```diff
@@ -53,7 +53,7 @@
         ".F..==========o==================o==========..F.",
         ".F..==========o==================o==========..F.",
         "FFG..........................................GFF",
-        "FTFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFTF",
+        "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF",
```
Comment (kevingranade) [inline data/json/mapgen/helipad.json:56]:
```suggestion
        "FTFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFTF",
```

File: data/json/mapgen/helipad.json [hunk 4]
```diff
@@ -117,7 +117,6 @@
       },
       "vendingmachines": { "1": { "item_group": "vending_drink", "lootable": true }, "2": { "item_group": "vending_food", "lootable": true } },
       "place_monsters": [ { "monster": "GROUP_MIL_STRONG", "x": 36, "y": 15 }, { "monster": "GROUP_MIL_WEAK", "x": 24, "y": 30, "density": 2 } ],
```
Comment (kevingranade) [inline data/json/mapgen/helipad.json:119]:
```suggestion
      "place_monsters": [ { "monster": "GROUP_MIL_STRONG", "x": 36, "y": 15 }, { "monster": "GROUP_MIL_WEAK", "x": 24, "y": 30, "density": 2 } ],
      "monster": { "T": { "monster": "mon_turret_spotlight" } },
```

File: data/json/mapgen/helipad.json [hunk 5]
```diff
@@ -117,7 +117,7 @@
...
-      "monster": { "T": { "monster": "mon_turret_rifle" } },
+      "monster": { "T": { "monster": "mon_turret_spotlight" } },
```
Comment (kevingranade) [inline data/json/mapgen/helipad.json:120]:
```suggestion
      "monster": { "T": { "monster": "mon_turret_searchlight" } },
```

**Pull request merged**